packages/lodestar-validator/src/api/interface/validators.ts

```diff
@@ -45,11 +45,13 @@ export interface IValidatorApi {
   publishAttestation(attestation: Attestation): Promise<void>;
 
   publishAggregatedAttestation(
-    aggregated: Attestation, validatorPubKey: BLSPubkey, slotSignature: BLSSignature
```

Comment:
I'm uneasy about using ValidatorIndex here (instead of BLSPubkey) but not changing it everywhere else.

If the validator has access to its ValidatorIndex, the entire API can likely be reworked rely on it (instead of mapping from BLSPubkey).
VC  -> getValidator(pubkey) -> BN
VC  <- {validatorIndex} <- BN
VC -> subscribeToEvents(validatorIndex) -> BN
VC <- {events} <- BN
VC -> publishData(data[, validatorIndex]) -> BN

Reply:
I see, this is a general beacon API. Can we use the same shape lighthouse is using? https://lighthouse-book.sigmaprime.io/http_beacon.html#beaconvalidators